include/seqan3/core/detail/to_string.hpp

```diff
@@ -0,0 +1,35 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \author Svenja Mehringer <svenja.mehringer AT fu-berlin.de>
+ * \brief Auxiliary for pretty printing of exception messages.
+ */
+
+#pragma once
+
+#include <seqan3/io/stream/debug_stream.hpp>
+
+namespace seqan3::detail
+{
+
+/*!\brief Streams all parameters via the seqan3::debug_stream and returns a concatenated string.
+ *
+ * \tparam    value_type Must be streamable (stream << value).
```

Comment:
Don't we want a `requires` for that?

Reply:
Yes but we would need to require that the type is streamable via the debug_stream_type which currently always resolves to true anyway because it has a generic, unconstrained overload :( I tried to changes this which causes a ICE on the rangesv3 1.0 beta branch (see #710)

Reply:
ok